Global/EnemyAction.gd

```diff
@@ -0,0 +1,29 @@
+extends RefCounted
```

Comment:
How does RefCounted differ from Resource?

Reply:
https://docs.godotengine.org/en/stable/tutorials/best_practices/node_alternatives.html

Per my understanding the main difference is Resource adds the ability to be serialized/deserialized which allows you to export properties into/from the inspector. If I'm not misunderstanding or missing something then I do not really think we need anything from Resource specifically ?

Reply:
Yeah, I guess if we don't need any export properties then this is fine